tutorials/11-p2p-shared-kv-cache.md

```diff
@@ -0,0 +1,184 @@
+# Tutorial: P2P KV Cache Sharing with LMCache
+
+## Introduction
+
+This tutorial demonstrates how to enable P2P KV cache sharing using LMCache in a vLLM deployment. KV cache will be retrieved from another vLLM pod if the KV cache does not exist locally.
+vLLM Production Stack uses LMCache for P2P KV cache sharing. For more details, see the [LMCache GitHub repository](https://github.com/LMCache/LMCache).
+
+## Table of Contents
+
+1. [Prerequisites](#prerequisites)
+2. [Step 1: Configuring P2P KV Cache Storage](#step-1-configuring-p2p-kv-cache-shared-storage)
+3. [Step 2: Deploying the Helm Chart](#step-2-deploying-the-helm-chart)
+4. [Step 3: Verifying the Installation](#step-3-verifying-the-installation)
+5. [Benchmark the Performance Gain of P2P Shared Storage (Work in Progress)](#benchmark-the-performance-gain-of-p2p-shared-storage-work-in-progress)
+
+## Prerequisites
+
+- Completion of the following tutorials:
+  - [00-install-kubernetes-env.md](00-install-kubernetes-env.md)
+  - [01-minimal-helm-installation.md](01-minimal-helm-installation.md)
+  - [02-basic-vllm-config.md](02-basic-vllm-config.md)
+- A Kubernetes environment with GPU support.
+
+## Step 1: Configuring P2P KV Cache Shared Storage
+
+Locate the file `tutorials/assets/values-11-p2p-sharing.yaml` with the following content:
+
+```yaml
+servingEngineSpec:
+  runtimeClassName: ""
+  modelSpec:
+  - name: "mistral-1"
+    repository: "lmcache/vllm-openai"
+    tag: "latest"
+    modelURL: "mistralai/Mistral-7B-Instruct-v0.2"
+    replicaCount: 2
+    requestCPU: 10
+    requestMemory: "40Gi"
+    requestGPU: 1
+    pvcStorage: "50Gi"
+    vllmConfig:
+      enableChunkedPrefill: false
+      enablePrefixCaching: false
+      maxModelLen: 16384
+
+    lmcacheConfig:
+      enabled: true
+      cpuOffloadingBufferSize: "20"
+      p2pEnabled: true
+      distributedPort: "8200"
+
+    hf_token: <YOUR HF TOKEN>
+
+  - name: "mistral-2"
+    repository: "lmcache/vllm-openai"
+    tag: "latest"
+    modelURL: "mistralai/Mistral-7B-Instruct-v0.2"
+    replicaCount: 2
+    requestCPU: 10
+    requestMemory: "40Gi"
+    requestGPU: 1
+    pvcStorage: "50Gi"
+    vllmConfig:
+      enableChunkedPrefill: false
+      enablePrefixCaching: false
+      maxModelLen: 16384
+
+    lmcacheConfig:
+      enabled: true
+      cpuOffloadingBufferSize: "20"
+      p2pEnabled: true
+      distributedPort: "8201"
+
+    hf_token: <YOUR HF TOKEN>
+
+lookupserverSpec:
+  replicaCount: 1
+  containerPort: 8080
```

Comment:
Redis by default uses port 6379. There is nothing in the lookup-server's deployment that changes that, therefore this should be changed.